docs-guides/source/mlmodel-utilities.md

```diff
@@ -209,3 +209,51 @@ ct.models.utils.bisect_model(
     merge_chunks_to_pipeline=True,
 )
 ```
+
+## Change Model Tensor Input/Output Types
+
+Consider a scenario when we have a Core ML model with an fp32 multiarray output, but we need to use a Core ML API that
+requires fp16 multiarrays instead. We can now easily change the model output types from fp32 to fp16 (and vice versa).
+
+An example how to update the output data types:
+
+```python
+from coremltools.models.model import MLModel
+from coremltools.utils import change_array_output_type
+from coremltools.proto.FeatureTypes_pb2 import ArrayFeatureType
+
+model = MLModel("my_model.mlpackage")
+updated_model = change_input_output_tensor_type(
+    ml_model=model,
+    from_type=ArrayFeatureType.FLOAT32,
+    to_type=ArrayFeatureType.FLOAT16,
+)
+updated_model.save("my_updated_model.mlpackage")
+```
+
+Another example is showing how to update data types of all the function inputs:
+```python
+from coremltools.models.model import MLModel
+from coremltools.utils import change_array_output_type
+from coremltools.proto.FeatureTypes_pb2 import ArrayFeatureType
+
+model = MLModel("my_model.mlpackage")
+updated_model = change_input_output_tensor_type(
+    ml_model=model,
+    from_type=ArrayFeatureType.FLOAT32,
+    to_type=ArrayFeatureType.FLOAT16,
+    function_names=["main_1", "main_2"],
+    input_names=["*"],
+    output_names=[],  # no output to be modified
+)
+updated_model.save("my_updated_model.mlpackage")
+```
+
+Optional arguments:
+* `function_names`: list of functions to be modified (by default only input / output of the `main` function is modified)
+* `input_names`: list of inputs that should be updated (by default none is modified)
```

Comment:
Any specific reason about the different default behaviors of `input_names` (none) and `output_names` (all)?

Reply:
It was meant to support the original use case (only to update all the output types) by default. But I'm open to any other default settings if you think those would be better.

Reply:
As the utility name is `change_input_output_tensor_type`, I think it might be better to keep same default behavior for input and output?

Reply:
Actually it's fine, let's revisit that in the future if needed.